### PR TITLE
Daniel/enrolment journey

### PR DESCRIPTION
**Pull request by DanielConnelly**

### CODE REVIEW

File: app/connectors/VPLAuthConnector.scala [hunk 1]
```diff
@@ -19,21 +19,81 @@ package connectors
...
+      }
+
+  private def getAuthority[A: Reads](implicit hc: HeaderCarrier) =
+    http.GET[JsValue](s"$serviceUrl/auth/authority").map(_.as[A])
+
+  def userDetails[A](ctx: A)(implicit hc: HeaderCarrier): Future[UserDetails] = ctx match {
+      case x: AuthContext =>
+        for {
+          ids       <- this.getIds[ExternalId](x)
+          userInfo  <- this.getUserDetails[JsValue](x)
```
Comment (DanielConnelly) [inline app/connectors/VPLAuthConnector.scala]: Remove these lines

File: app/models/enrolment/EnrolmentAccountIndividual.scala [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    val email = "email"
+    val confirmedEmail = "confirmedEmail"
+  }
+
+  lazy val form = Form(mapping(
+    keys.firstName -> nonEmptyText,
+    keys.lastName -> nonEmptyText,
+    keys.address -> addressMapping,
+    keys.phone -> nonEmptyText(maxLength = 15),
+    keys.mobilePhone -> optional(text(maxLength = 15)), //Pretty sure this is meant to be optional.
```
Comment (DanielConnelly) [inline app/models/enrolment/EnrolmentAccountIndividual.scala]: remove comment

File: app/models/enrolment/EnrolmentAccountIndividual.scala [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+object FieldData {
+
+  def apply(userInfo: UserInfo): FieldData =
+    new FieldData(
+      firstName = userInfo.firstName,
+      lastName = userInfo.lastName,
+      postcode = userInfo.postcode,
+      email = userInfo.email)
+
+  implicit private def get(opt: Option[String]): String = opt match {
```
Comment (DanielConnelly) [inline app/models/enrolment/EnrolmentAccountIndividual.scala]: Delete this

File: app/models/enrolment/EnrolmentAccountOrganisation.scala [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+case class EnrolmentOrganisationAccountDetails(firstName: String,
+                                               lastName: String,
+                                               companyName: String,
+                                               address: Address,
+                                               phone: String,
+                                               email: String,
+                                               confirmedEmail: String,
+                                               isAgent: Boolean)
+
+//Do we really need two classes that take in the same data ???
```
Comment (DanielConnelly) [inline app/models/enrolment/EnrolmentAccountOrganisation.scala]: Remove unneed class

File: app/views/createAccount/group.scala.html [hunk 5]
```diff
@@ -14,107 +14,107 @@
...
+                    field = model.form(keys.isAgent),
+                    options = Seq("true", "false"),
+                    dataAttributes = Nil,
+                    args = Seq(
+                        '_label -> Messages(s"label.createAccount.${keys.isAgent}"),
+                        '_labelClass -> "form-label-bold",
+                        '_legend -> true,
+                        '_fieldsetClass -> "inline"
+                    )
+                )
```
Comment (DanielConnelly) [inline app/views/createAccount/group.scala.html]: check this hasnt changed

File: test/controllers/CreateEnrolmentUserSpec.scala [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+      "email" -> Seq("x@x.com"),
+      "confirmedBusinessEmail" -> Seq("x@x.com"),
+      "isAgent" -> Seq("false")
+    )
+    val fakeRequest: FakeRequest[AnyContent] = FakeRequest().withBody(AnyContentAsFormUrlEncoded(data))
+    val res = TestCreateEnrolmentUser.submitOrganisation()(fakeRequest)
+    status(res) mustBe SEE_OTHER
+  }
+
+  /*"Submitting a valid form" should "redirect to SIV" in {
```
Comment (DanielConnelly) [inline test/controllers/CreateEnrolmentUserSpec.scala]: Delete SIV tests

File: app/actions/AuthenticatedAction.scala [hunk 7]
```diff
@@ -86,21 +94,60 @@ class AuthenticatedAction @Inject()(provider: GovernmentGatewayProvider,
   private def handleResult(result: AuthorisationResult, body: BasicAuthenticatedRequest[AnyContent] => Future[Result])
                           (implicit request: Request[AnyContent]) = {
     result match {
-      case Authenticated(accounts) => body(BasicAuthenticatedRequest(accounts.organisation, accounts.person, request))
+      case Authenticated(accounts) => success(accounts, body)
       case InvalidGGSession => provider.redirectToLogin
-      case NoVOARecord => Future.successful(Redirect(controllers.routes.CreateIndividualAccount.show))
+      case NoVOARecord => // once enrolment enabled we need to do this with the nonOrganisationAction | NonOrganisationAccount =>
```
Comment (DanielConnelly) [inline app/actions/AuthenticatedAction.scala]: DI in implementation for enrolment

File: app/actions/AuthenticatedAction.scala [hunk 8]
```diff
@@ -86,26 +94,93 @@ class AuthenticatedAction @Inject()(provider: GovernmentGatewayProvider,
...
+  override def success(
+                        accounts: Accounts,
+                        body: BasicAuthenticatedRequest[AnyContent] => Future[Result])
+                      (implicit request: Request[AnyContent]): Future[Result] = {
+    def handleError: PartialFunction[Throwable, Future[Result]] = {
+      case _: InsufficientEnrolments =>
+        enrolments.enrol(accounts.person.individualId, accounts.organisation.addressId).flatMap {
+          case Success =>
+            Future.successful(Ok(views.html.createAccount.migration_success()))
+          //TODO use ^ this one instead of the following ???
```
Comment (peter-hazell) [inline app/actions/AuthenticatedAction.scala]: comment!

**Pull request merged**